index/postings_test.go

```diff
@@ -541,3 +698,81 @@ func TestIntersectWithMerge(t *testing.T) {
 	testutil.Ok(t, err)
 	testutil.Equals(t, []uint64{30}, res)
 }
+
+func TestWithoutPostings(t *testing.T) {
+	var cases = []struct {
+		base Postings
+		drop Postings
+
+		res Postings
+	}{
+		{
+			base: EmptyPostings(),
+			drop: EmptyPostings(),
+
+			res: EmptyPostings(),
+		},
+		{
+			base: EmptyPostings(),
+			drop: toPostings(1, 2),
+
+			res: EmptyPostings(),
+		},
+		{
+			base: toPostings(1, 2),
+			drop: EmptyPostings(),
+
+			res: toPostings(1, 2),
+		},
+		{
+			base: toPostings(),
+			drop: toPostings(),
+
+			res: toPostings(),
+		},
+		{
+			base: toPostings(1, 2, 3),
+			drop: toPostings(),
+
+			res: toPostings(1, 2, 3),
+		},
+		{
+			base: toPostings(1, 2, 3),
+			drop: toPostings(4, 5, 6),
+
+			res: toPostings(1, 2, 3),
+		},
+		{
+			base: toPostings(1, 2, 3),
+			drop: toPostings(3, 4, 5),
+
+			res: toPostings(1, 2),
+		},
+	}
+
+	for _, c := range cases {
+		t.Run("", func(t *testing.T) {
+			if c.res == nil {
```

Comment:
don't think we need this. A panic is ok in a test in my opinion

Reply:
If you think so, can remove